### PR TITLE
Manual for Remote Execution 1.3.3

### PR DESCRIPTION
**Pull request by adamruzicka**
- [ ] figure out how to convert webm recording into a gif which is not bad or huge (or both)

### CODE REVIEW

File: plugins/foreman_remote_execution/1.3/index.md [hunk 1]
```diff
@@ -0,0 +1,877 @@
...
+
+Due to various reasons, dynflow was separated from Smart Proxy into a standalone
+service called smart_proxy_dynflow_core. Most of the gems originally required for
+remote execution also followed this example and were split into regular and -core part.
+The original gems are usually either Foreman or Smart Proxy plugins, the -core parts
+are required by the smart proxy dynflow core service.
+
+This split approach also allows us to simplify the deployment by loading the -core gems
+into Smart Proxy or even directly into Foreman and thus lowering the entry barrier
+at the cost of lower performance.
```
Comment (mmoll) [inline plugins/foreman_remote_execution/1.3/index.md]: I think this is not interesting for users.

File: plugins/foreman_remote_execution/1.3/index.md [hunk 2]
```diff
@@ -0,0 +1,877 @@
...
+
+Please note that on Debian Dynflow will be set up as part of the Foreman Proxy. On other
+platforms it will be a standalone service.
```
Comment (mmoll) [inline plugins/foreman_remote_execution/1.3/index.md]: comma missing after "Debian"

Comment (mmoll) [reply plugins/foreman_remote_execution/1.3/index.md]: Also, isn't EL7 the only only left of "other platforms"? If yes, that whole paragraph should be reworded.

File: plugins/foreman_remote_execution/1.3/index.md [hunk 3]
```diff
@@ -0,0 +1,877 @@
...
+
+### With Foreman Installer
+
+You can install both the Foreman and Smart Proxy plugin with the installer:
+
+    foreman-installer --enable-foreman-plugin-remote-execution\
+                      --enable-foreman-proxy-plugin-remote-execution-ssh
+
+The installer will automatically configure an SSH key for you, or you may use an existing see.
+See the `foreman-installer --help` for more information.
```
Comment (mmoll) [inline plugins/foreman_remote_execution/1.3/index.md]: minus "the"

File: plugins/foreman_remote_execution/1.3/index.md [hunk 4]
```diff
@@ -0,0 +1,877 @@
...
+
+    apt-get install ruby-foreman-remote-execution\
+                    ruby-smart-proxy-remote-execution-ssh
+
+The next thing to do is make sure the Smart Proxy, Smart Proxy Dynflow Core
+and Foreman can talk to each other. If you're on Debian, there's no need to do anything.
+This is done by properly configuring `:core_url` in `/etc/foreman-proxy/settings.d/dynflow.yml` and `:foreman_url`, `:listen`
+and `:port` keys in `/etc/smart_proxy_dynflow_core/settings.yml`.
+
+Next you have to setup ssh keys. By default smart proxy loads the key
```
Comment (mmoll) [inline plugins/foreman_remote_execution/1.3/index.md]: "set up"

File: plugins/foreman_remote_execution/1.3/index.md [hunk 5]
```diff
@@ -0,0 +1,877 @@
...
+from `/usr/share/foreman-proxy/.ssh/id_rsa_foreman_proxy`. To customize it you
+can edit the configuration in
+`/etc/foreman-proxy/config/settings.d/remote_execution_ssh.yml`. Without
+customization you need to create new ssh key and distribute it to
```
Comment (mmoll) [inline plugins/foreman_remote_execution/1.3/index.md]: "...create **a** new ssh key"

File: plugins/foreman_remote_execution/1.3/index.md [hunk 6]
```diff
@@ -0,0 +1,877 @@
...
+target hosts. The key must not use passphrase.
```
Comment (mmoll) [inline plugins/foreman_remote_execution/1.3/index.md]: I think better would be something like "...must not be protected by a passphrase."

File: plugins/foreman_remote_execution/1.3/index.md [hunk 7]
```diff
@@ -0,0 +1,877 @@
...
+Note: If you don't see a menu entry, the Foreman plugin is not installed
+correctly.
+
+Now you should see a form similar to the Provisioning template one. In first
+tab you have to fill in **Name**. You can use input macros in ERB tags which
+stands for variables which you define in second tab called Job.
+
+![Job Template form](/plugins/foreman_remote_execution/{{page.version}}/job_template_form.png)
+
+Before we get to inputs, lets look at other fields in Job tab. First one is an
```
Comment (mmoll) [inline plugins/foreman_remote_execution/1.3/index.md]: "let's"

File: plugins/foreman_remote_execution/1.3/index.md [hunk 8]
```diff
@@ -0,0 +1,877 @@
...
+per job invocation as well so this is just a default value for particular
+job template. Two extra placeholders `%{template_name}` and `%{job_category}`
+can be used.
+
+On the very same tab you can configure inputs. These template inputs
+defines values that can be accessed in the template itself. To add new input,
+click the green button "+ Add Input" which adds more fields to the form. To
+remove the input, hit the cross in top right corner of the input form.
+
+We support four types of inputs. Each must have it's **name** which is
```
Comment (mmoll) [inline plugins/foreman_remote_execution/1.3/index.md]: "its"

File: plugins/foreman_remote_execution/1.3/index.md [hunk 9]
```diff
@@ -0,0 +1,877 @@
...
+Alternatively, you may enable a fallback mode by setting
+**remote_execution_fallback_proxy** to true.  In this case, the plugin will
+consider any proxy associated with the host, such as its Puppetmaster, as long
+as the proxy also has a remote execution feature.
+
+## 4.2 Global settings
+
+Under Administer -> Settings you can find RemoteExecution tab which allows you
+to customize plugin behavior. Here is the list of settings and their meaning.
+
```
Comment (mmoll) [inline plugins/foreman_remote_execution/1.3/index.md:634]: for whatever reason in my rendered version theres a "</td>" as text there

File: plugins/foreman_remote_execution/1.3/index.md (hunk 4 above)
Comment (adamruzicka) [reply plugins/foreman_remote_execution/1.3/index.md]: just "next you have to set up ssh keys" or "next you have to set ssh keys up"?

Comment (mmoll) [reply plugins/foreman_remote_execution/1.3/index.md]: "next you have to set up ssh keys"

**Review by dLobatog — APPROVED** [review]
Thanks @adamruzicka @mmoll

**Pull request merged**